src/stactools/landsat/stac.py

```diff
@@ -1,102 +1,281 @@
+            if eo_band is not None:
+                asset = item.assets[key]
+                eo_asset = EOExtension.ext(asset, add_if_missing=True)
+                eo_asset.bands = [Band.create(**eo_band)]
+            raster_band = raster_bands.get(key, None)
+            if raster_band is not None:
+                asset = item.assets[key]
+                raster = RasterExtension.ext(asset, add_if_missing=True)
+                raster.bands = [RasterBand.create(**raster_band)]
+
+        # -- Thermal assets (only exist if optical exist)
+        if mtl_metadata.processing_level == 'L2SP':
+            assets = copy.deepcopy(SENSOR_ASSETS[sensor_key]["ST"])
+            eo_bands = EO_BANDS[sensor_key]["ST"]
+            raster_bands = RASTER_BANDS[sensor_key]["ST"]
+            for key, asset_dict in assets.items():
+                asset_dict["type"] = MediaType.COG
+                asset_dict["href"] = f"{base_href}_{key}.TIF"
+                item.add_asset(key, Asset.from_dict(asset_dict))
+                eo_band = eo_bands.get(key, None)
+                if eo_band is not None:
+                    asset = item.assets[key]
+                    eo_asset = EOExtension.ext(asset, add_if_missing=True)
+                    eo_asset.bands = [Band.create(**eo_band)]
+                raster_band = raster_bands.get(key, None)
+                if raster_band is not None:
+                    asset = item.assets[key]
+                    raster = RasterExtension.ext(asset, add_if_missing=True)
+                    raster.bands = [RasterBand.create(**raster_band)]
+
+        # -- USGS link
+        instrument = "-".join(i for i in SENSORS[sensor.value]["instruments"])
+        usgs_item_page = (
+            f"https://landsatlook.usgs.gov/stac-browser/collection02/level-"
+            f"{level}/standard/{instrument}"
+            f"/{mtl_metadata.scene_datetime.year}"
+            f"/{mtl_metadata.wrs_path}/{mtl_metadata.wrs_row}"
+            f"/{mtl_metadata.product_id}")
+
+        item.add_link(
+            Link(rel="alternate",
+                 target=usgs_item_page,
+                 title="USGS stac-browser page",
+                 media_type="text/html"))
 
     return item
+
+
+def get_usgs_geometry(
+    base_href: str,
+    sensor: Sensor,
+    product_id: str,
+    read_href_modifier: Optional[ReadHrefModifier] = None
+) -> Union[Dict[str, Any], None]:
+    """Attempts to get scene geometry from a USGS STAC Item.
+
+    Args:
+        base_href (str): Base href to the planetary computer storage
+        sensor (Sensor): Enum of MSS, TM, ETM, or OLI-TIRS
+        product_id (str): Scene product id from mtl metadata
+        read_href_modifier (Callable[[str], str]): An optional function to
+            modify the MTL STAC href (e.g. to add a token to a url)
+    Returns:
+        Union[Dict[str, Any], None]: Either a GeoJSON geometry or None
+    """
+    # Check Planetary Computer storage first
```

Comment:
How is this defaulting to Planetary Computer? `base_href` isn't optional, so it always has to be passed into the function.

Reply:
Yep. Corrected.